ci/build/build-release.sh

```diff
@@ -153,4 +145,34 @@ EOF
   popd
 }
 
+create_shrinkwraps() {
+  # yarn.lock or package-lock.json files (used to ensure deterministic versions of dependencies) are
+  # not packaged when publishing to the NPM registry.
+  # To ensure deterministic dependency versions (even when code-server is installed with NPM), we create
+  # an npm-shrinkwrap.json file from the currently installed node_modules. This ensures the versions used
+  # from development (that the yarn.lock guarantees) are also the ones installed by end-users.
+
+  # We first generate the shrinkwrap file for code-server itself
+  npm shrinkwrap
+  # HACK@edvincent: The shrinkwrap file will contain the devDependencies, which by default
+  # are installed if present in a lockfile. To avoid every user having to specify --production
+  # to skip them, we carefully remove them from the shrinkwrap file.
+  json -f npm-shrinkwrap.json -I -e "Object.keys(this.dependencies).forEach(dependency => { if (this.dependencies[dependency].dev) { delete this.dependencies[dependency] } } )"
+  mv npm-shrinkwrap.json "$RELEASE_PATH"
+
+  # Then the shrinkwrap files for the bundled VSCode
+  # We don't need to remove the devDependencies for these because we control how it's installed - and
+  # as such we can force the --production flag
+  cd lib/vscode/
+  npm shrinkwrap
+  json -f npm-shrinkwrap.json -I -e "Object.keys(this.dependencies).forEach(dependency => { if (this.dependencies[dependency].dev) { delete this.dependencies[dependency] } } )"
```

Comment:
My bad; I should have made a note of this earlier; we recently switching to using `lib/vscode/remote/package.json` since it turns out the upper `package.json` is for desktop VS Code and one in `remote` is for the web version (which is what code-server is running).

Could we generate the shrinkwraps directly in `$RELEASE_PATH`?  That way it is always using the right `package.json` no matter what we end up copying into it.

Reply:
Yup let me try that. For now mostly trying to get it to build more than anything, so no probs on other things.